x-pack/plugins/security/server/routes/api/public/roles/put.js

```diff
@@ -78,8 +79,8 @@ export function initPutRolesApi(
       run_as: Joi.array().items(Joi.string()),
     }),
     kibana: Joi.object().keys({
-      global: Joi.array().items(Joi.string().valid(Object.keys(privilegeMap))),
-      space: Joi.object().pattern(/^/, Joi.array().items(Joi.string().valid(Object.keys(privilegeMap))))
+      global: Joi.array().items(Joi.string().valid(Object.keys(privilegeMap.global))),
+      space: Joi.object().pattern(/^/, Joi.array().items(Joi.string().valid(Object.keys(privilegeMap.space))))
```

Comment:
I think we'll need to update the Joi object pattern used here. The way it's currently written (and I know you didn't change this), the API would allow a key of `*` or similar, which would end up granting access to a bucket load of spaces